ipfs-cluster/trigger-server/triggerserver.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+    "log"
+    "net/http"
+    "os/exec"
+    "time"
+    "fmt"
+    "io/ioutil"
+    "math/rand"
+    "strconv"
+    "os"
+    "github.com/ipfs/go-ipfs-api"
+    "bytes"
+)
+
+const CONFIG_PATH = "/etc/trigger-server"
+// HTTP API listen port
+const PORT = 8082
+var RUNNER_PATH = os.Getenv("GOPATH") + "src/github.com/ipfs/kubernetes-ipfs/ipfs-cluster/"
+// Runner takes two args
+const RUNNER_NUM_NODES = "5"
+const RUNNER_NUM_PINS = "5"
+
+// Secret is set by main
+var secret = ""
+
+// IPFS Shell
+var sh = shell.NewShell("localhost:5001")
+
+func run(rw http.ResponseWriter, req *http.Request) {
+    if req.Method == "POST" {
+	if req.Header.Get("X-Auth") != secret {
+		log.Println("User failed authentication")
+		fmt.Fprintf(rw, "Invalid authentication\n")
+	} else {
+		// This command takes a while, so we'll hang out here while it runs
+		cmd := exec.Command("bash", "runner.sh", RUNNER_NUM_NODES, RUNNER_NUM_PINS)
+		cmd.Dir = RUNNER_PATH
+		out, err := cmd.CombinedOutput()
+		// Add the output to IPFS
+		addr, ipfsErr := sh.Add(bytes.NewReader(out))
+		// Make it an IPFS link
+		addr = "/ipfs/" + addr
+		// Send the link to the client
+		fmt.Fprintf(rw, addr)
+		// Handle errors that may have occurred
+		if err != nil {
+			log.Println("Error from runner")
+			log.Println(err)
+		}
+		if ipfsErr != nil {
+			log.Println("IPFS Error adding output")
+			log.Println(ipfsErr)
+		}
+	}
+    } else {
+	// For non-POST requests, tell the user what we are.
+        fmt.Fprintf(rw, `Kubernetes-IPFS Trigger Server
+	https://github.com/ipfs/kubernetes-ipfs/
+	`)
+    }
+}
+
+func main() {
+    secret_bytes, readErr := ioutil.ReadFile(CONFIG_PATH)
+    if readErr != nil {
+        log.Printf("Missing secret configuration in %s. We'll generate one for you...\n", CONFIG_PATH)
+	rand.Seed(time.Now().UnixNano())
```

Comment:
customary to initialize seeds in init(). Might as well use a proper random source from `crypto` package. Just need to read enough bytes into a buffer and SHA it for example.